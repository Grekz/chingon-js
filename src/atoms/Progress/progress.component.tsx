import styled from '@emotion/styled'
import * as React from 'react'

interface IProgressProps {
  progress: number
  max: number
  color: 'string'
  backColor: 'string'
  children?: React.ReactNode
}

const Progress = (props: IProgressProps) => (
  <ProgressWrapper {...props}>
    <div>{props.children}</div>
  </ProgressWrapper>
)

Progress.defaultProps = {
  backColor: '#e9ecef',
  color: '#007bff',
  max: 100,
  progress: 30,
}

const ProgressWrapper = styled.div`
  display: flex;
  height: 1rem;
  overflow: hidden;
  font-size: 0.75rem;
  background-color: ${({ backColor }: IProgressProps) => backColor};
  border-radius: 0.25rem;
  > div {
    display: -ms-flexbox;
    display: flex;
    -ms-flex-direction: column;
    flex-direction: column;
    -ms-flex-pack: center;
    justify-content: center;
    color: #fff;
    text-align: center;
    white-space: nowrap;
    background-color: ${({ color }: IProgressProps) => color};
    transition: width 0.6s ease;
    width: ${({ progress, max }: IProgressProps) =>
      `${(progress / max) * 100}%`};
  }
`

export default Progress
