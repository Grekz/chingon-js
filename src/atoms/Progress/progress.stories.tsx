import * as React from 'react'

// tslint:disable-next-line: no-implicit-dependencies
import { storiesOf } from '@storybook/react'

import Progress from '.'

storiesOf('Atoms|Progress', module).add('As Default', () => (
  <Progress max={100} progress={20} />
))
