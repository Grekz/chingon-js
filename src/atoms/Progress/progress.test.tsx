import { configure, shallow } from 'enzyme'
import * as Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'
import serializer, { matchers } from 'jest-emotion'
import * as React from 'react'
import Progress from '.'

configure({ adapter: new Adapter() })
expect.extend(matchers)
expect.addSnapshotSerializer(serializer)

describe('<Progress /> rendering', () => {
  it('should render Progress', () => {
    const wrapper = shallow(<Progress max={100} progress={20} />)
    expect(wrapper).toHaveLength(1)
  })
  it('should match snapshot', () => {
    const tree = shallow(<Progress max={100} progress={20} />)
    expect(toJson(tree)).toMatchSnapshot()
  })
})
