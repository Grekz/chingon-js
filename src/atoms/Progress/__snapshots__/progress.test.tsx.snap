// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Progress /> rendering should match snapshot 1`] = `
<Styled(div)
  backColor="#e9ecef"
  color="#007bff"
  max={100}
  progress={20}
>
  <div />
</Styled(div)>
`;
